files/image_config/monit/process_checker

```diff
@@ -27,20 +29,61 @@ def check_process_existence(container_name, process_cmdline):
             # We leveraged the psutil library to help us check whether the process is running or not.
             # If the process entity is found in process tree and it is also in the 'running' or 'sleeping'
             # state, then it will be marked as 'running'.
+
+            # For given feature we get the host and network namespace instances it's processes should be running
+            # based on it's scope and add it to expected set.
+
+            # From psutil we get number of running instances of the processes and add it to the the actual set
+
+            # Difference bwetween expected and actual set provides instances where the processes are not running
+            # and will be logged as syslog message by monit
+
+            process_namespace_expected_set = set()
+            process_namespace_found_set = set()
+
+            has_global_scope = ast.literal_eval(feature_table[container_name].get('has_global_scope', 'True'))
+            has_per_asic_scope = ast.literal_eval(feature_table[container_name].get('has_per_asic_scope', 'False'))
+
+            if has_global_scope:
+                process_namespace_expected_set.add(multi_asic.DEFAULT_NAMESPACE)
+
+            if has_per_asic_scope:
+               process_namespace_expected_set.update(multi_asic.get_namespace_list())
```

Comment:
Is it possible for both to be True, or both to be False?

Reply:
One more question is the value of `multi_asic.DEFAULT_NAMESPACE` is `''` string and the function `get_namespace_list()` will return a list which includes each ASIC name such as `asic1`, right?

Reply:
> Is it possible for both to be True, or both to be False?

@jleveque Yes both can be true . For now lldp is the use-case. Both can not be False there is no use-case as of now for that.

Reply:
> One more question is the value of `multi_asic.DEFAULT_NAMESPACE` is `''` string and the function `get_namespace_list()` will return a list which includes each ASIC name such as `asic1`, right?

@yozhao101 yes, this is correct.

Reply:
Thanks @abdosi!

Reply:
Does the line 51 have a indentation issue?

Reply:
@yozhao101 Updated.